### ISSUE TITLE
Fix Travis

### ISSUE THREAD

**Issue by mkurz** [playframework/play-meta#185]
Look into what's going on with Travis and make it work again.

**Comment by ennru** [comment]
Lightbend keeps paying for the 1 concurrent Travis worker subscription (USD 69/month).
It's not enough to get a workable turn-around with Travis, but the number of variations Travis builds can be reduced and migrating to GitHub's CI services seems like a good path anyway.

**Comment by mkurz** [comment]
I had a look at Travis once more...
First off all: Travis [does not offer concurrent plans anymore since Dec. 13th 2021](https://blog.travis-ci.com/2021-12-01-pricingenhancements) (yes just 4 days ago) - but if you are currently on a concurrent plan you can continue to use it (for a while at least). They now only offer usage-based pricing (which [was announced last year already](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing)), where you purchase credits and for this credits you get minutes (on a standard Linux machine for 10 credits you get 1 minutes). This [has an advantage:](https://docs.travis-ci.com/user/billing-overview/#usage-based-plans)

> ...organizations can run as many build job as they want simultaneously, meaning that all builds are executed as soon as possible without limitations.

So that should speed up builds, but probably is more expensive if you run many builds, because you need more minutes.

Anyway, I tried to find out what happend to our specific Travis account for Play, because we are still on a concurrent plan and until September we could run 5 concurrent jobs, but then problems started...
Either Lightbend paid for 2 concurrent jobs and Travis added 3 additional because Play is a OSS project (this is an old screenshot someone posted on twitter):
<kbd>
<img src="https://pbs.twimg.com/media/ER9JcJUW4AEYj0m?format=jpg&name=large" width="600" />
</kbd>

Or because Play migrated from travis-ci.org to travis-ci.com, Travis support assigned us 5 concurrent jobs on .com. That's because historically OSS projects on travis-ci.org (which is not available anymore) got 5 concurrent jobs for free.

(Some related discussions: https://travis-ci.community/t/concurrency-for-open-source-projects/3827 and https://twitter.com/travisci/status/1234495074882420741)

If I remember correctly Lightbend stopped to pay the bills for Travis around September until Enno activated the 1 concurrent jobs plan. It seems because the plan was switched we somehow lost our 5 concurrent job OSS goodie (or +3 additional or whatever that was). I will contact support and ask them, because we are a OSS project, if they could either again at least assign us 3 additional concurrent jobs to our now-outdated concurrent job plan or at least provide us [OSS credits](https://docs.travis-ci.com/user/billing-overview/#credits-vs-oss-only-credits) (also see [here](https://docs.travis-ci.com/user/billing-faq/#what-if-i-am-building-open-source)) in case we switch to the new usage-based model.

However, it seems their support is not very reponsive, sometimes [takes them weeks for a reply](https://travis-ci.community/t/oss-only-credits-request-processing-time-estimate/10446) and they check the project [for certain requirements](https://discourse.julialang.org/t/travis-ci-changes-pricing-plans/49528/13), so they could easily decline Play because either Lightbend is still in the game or even because we are sponsored by OpenCollective. Also, I read stories that they give out this OSS credits just once, and then you have to again contact support if you run out of them to get more...

Anyway, let's see. If they don't answer within the next 1-2 days, I think we will should just to switch to the usage-based plan for a couple of weeks, at least that would speed up the build, because as said, there will not be a concurrent limit. And migrate to GH actions asap. (Unfortunately we can not purchase just more concurrent jobs anymore, since that option is gone since a couple of days...)

**Comment by ihostage** [comment]
> However, it seems their support is not very reponsive, sometimes takes them weeks for a reply and ...

In June I requested OSS credits for my GitHub organization and they answered me the same day. I'm lucky. 😄

**Comment by ihostage** [comment]
> Also, I read stories that they give out this OSS credits just once, and then you have to again contact support if you run out of them to get more...

@mkurz 
Yep, it's true. Support gave me 10K credits and when they are used up, I should request them again.

**Comment by mkurz** [comment]
> In June I requested OSS credits for my GitHub organization and they answered me the same day.

That gives hope.
BTW, I got an email from Open Collective where they approved us to be legit to hand us out virtual credit cards. I now requested one for Travis CI last night, hopefully we get it today.
(Each virtual card is bound to a service, you can't just pay what you want with it. If we want to pay another service we have to request another virtual card for it.)

**Comment by BillyAutrey** [comment]
Re: Moving to actions - thankfully @ihostage did some the work.  So, at the very least, we should be able to get some CI/CD results, and compare then to the GitHub Actions output.  Maybe we don't have to pay for Travis for very long.

**Comment by mkurz** [comment]
Alright, just a couple of minutes ago OpenCollective issued us a virtual card with a limit of $1000/month.
Even better news: Open Collective switched virtual card providers meaning many of the former restrictions on virtual cards no longer apply - they are no longer vendor-locked, so we can pay different services with just one card.

**Comment by ihostage** [comment]
> we can pay different services with just one card.

Good news 🥳

**Comment by mkurz** [comment]
Because I don't think it makes sense to stay on the 1 concurrent plan any longer, since it is extremly slow, I just tried to switch Travis to the new usage-based model and start with the USD 30 monthly plan, which gives us 50k credits and should allow us to run jobs in parallel without limits (I still hope we get free OSS credits, from what I have seen on the web usually they give 25k.)
Anyway, switching plan doesn't occur immediately, but only after the billing period of the current plan is over... which would be 1st of January... Of course this is unsatisfying, so I contacted support to switch us now.

**Comment by mkurz** [comment]
So, Travis support answered:
Starting from January 1st, we will be on the usage-based model. That means we pay USD 30,- and get 50k credits (=5000 build minutes). Plus they donated us additional 25k OSS credits (=2500 minutes) _once_, however we can again contact support to re-fill...
Until the new plan takes effect on January 1st they also allow us to run five concurrent jobs instead of just one (which is what we pay for actually right now), that is because, for whatever reason, they can not put us on the 50k credits plan immediately. Actually, having five concurrent jobs is what we had back in the good old days. In theorie, however, starting with Saturday, there shouldn't be any concurrent limit anymore at all. Let's see.
If we run out of minutes, I would be OK to switch to the 100k (=10000 minutes) plan for 60 USD, which is still cheaper what we pay right now (USD 69). However, switching to GitHub actions still has priority.
